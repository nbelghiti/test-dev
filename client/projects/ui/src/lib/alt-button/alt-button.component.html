<button
	*ngIf="!link; else isLink"
	class="c-alt-button"
	(click)="clicked.emit($event)"
>
	<ng-container [ngTemplateOutlet]="altButtonContent"></ng-container>
</button>

<ng-template #isLink>
	<a
		class="c-alt-button"
		[routerLink]="link"
	>
		<ng-container [ngTemplateOutlet]="altButtonContent"></ng-container>
	</a>
</ng-template>

<ng-template #altButtonContent>
	<div
		*ngIf="iconBefore"
		class="c-alt-button__icon c-alt-button__icon--before"
	>
		<ui-icon
			[name]="iconBefore"
			[size]="IconSize.small"
		>
		</ui-icon>
	</div>

	<span class="c-alt-button__text">
		<ng-content></ng-content>
	</span>

	<div
		*ngIf="iconAfter"
		class="c-alt-button__icon c-alt-button__icon--after"
	>
		<ui-icon
			[name]="iconAfter"
			[size]="IconSize.small"
		>
		</ui-icon>
	</div>
</ng-template>
