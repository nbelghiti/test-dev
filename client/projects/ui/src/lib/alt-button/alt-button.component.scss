// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: AltButton
// =============================================================================
.c-alt-button {
	position: relative;

	display: inline-flex;
	justify-content: center;
	align-items: center;
	width: 100%;
	min-width: 8rem;
	height: 100%;

	padding: 1.5rem 3rem;

	text-align: center;
	text-decoration: none;

	border: $border-width-base dashed $background-color-orange-78;
	color: $primary-color;

	transition: border-color $animation-base;

	cursor: pointer;

	&:hover {
		border-color: $primary-color;
	}

	&:focus {
		border-style: solid;
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-alt-button__icon {
	display: flex;

	font-size: 1.5rem;
	line-height: $line-height-base;

	color: currentColor;

	transition: color $animation-base;
}

.c-alt-button__text {
	display: inline-block;
	max-width: 30rem;
	overflow: hidden;

	font-family: $font-family-secondary;
	font-size: $font-size-1x-small;
	line-height: $line-height-base;
	text-decoration: underline;
	text-overflow: ellipsis;
	white-space: nowrap;

	color: currentColor;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-alt-button__icon--after {
	margin-left: 1rem;
}

.c-alt-button__icon--before {
	margin-right: 1rem;
}
