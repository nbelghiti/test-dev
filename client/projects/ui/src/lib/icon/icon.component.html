<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
