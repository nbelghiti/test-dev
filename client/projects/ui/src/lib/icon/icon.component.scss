// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
