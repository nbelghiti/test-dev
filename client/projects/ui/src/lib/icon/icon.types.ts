export enum IconName {
	archive = 'archive',
	arrowDown = 'arrow-down',
	arrowLeft = 'arrow-left',
	arrowRight = 'arrow-right',
	arrowRightAlt = 'arrow-alt-circle-right',
	arrows = 'arrows-alt-h',
	arrowUp = 'arrow-up',
	bell = 'bell',
	bellSlash = 'bell-slash',
	book = 'book',
	brackets = 'brackets',
	calendar = 'calendar',
	caretDown = 'caret-down',
	caretLeft = 'caret-left',
	caretRight = 'caret-right',
	caretUp = 'caret-up',
	check = 'check',
	checkCircle = 'check-circle',
	chevronDown = 'chevron-down',
	chevronLeft = 'chevron-left',
	chevronRight = 'chevron-right',
	chevronUp = 'chevron-up',
	clock = 'clock',
	clone = 'clone',
	close = 'times-circle',
	code = 'code',
	codeBranch = 'code-branch',
	cube = 'cube',
	download = 'download',
	edit = 'edit',
	excel = 'file-excel',
	exclamation = 'exclamation',
	exclamationCircle = 'exclamation-circle',
	expand = 'expand',
	externalLink = 'external-link-alt',
	file = 'file-alt',
	home = 'home',
	horizontalEllipsis = 'ellipsis-h',
	industry = 'industry',
	infoCircle = 'info-circle',
	invisible = 'eye-slash',
	marker = 'map-marker-alt',
	minus = 'minus',
	pdf = 'file-pdf',
	pen = 'pen',
	plus = 'plus',
	print = 'print',
	search = 'search',
	settings = 'cog',
	signOut = 'sign-out-alt',
	snowflake = 'snowflake',
	spinner = 'spinner',
	star = 'star',
	sync = 'sync-alt',
	table = 'table',
	times = 'times',
	trash = 'trash',
	upload = 'upload',
	user = 'user',
	users = 'users',
	visible = 'eye',
	plusSquare = 'plus-square',
	minusSquare = 'minus-square',
}

export enum IconType {
	success = 'success',
	error = 'error',
	warning = 'warning',
	info = 'info',
}

export enum IconSize {
	large = 'large',
	small = 'small',
	tiny = 'tiny',
}
