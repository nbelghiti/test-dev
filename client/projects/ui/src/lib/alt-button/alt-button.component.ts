import { Component, EventEmitter, Input, Output } from '@angular/core';

import { IconName, IconSize } from '../icon/icon.types';

@Component({
	selector: 'ui-alt-button',
	templateUrl: './alt-button.component.html',
	styleUrls: ['./alt-button.component.scss']
})

export class AltButtonComponent {
	@Input() public link?: string;
	@Input() public iconBefore?: IconName;
	@Input() public iconAfter?: IconName;
	@Output() public clicked: EventEmitter<MouseEvent> = new EventEmitter<MouseEvent>();

	public IconSize: typeof IconSize = IconSize;
}
