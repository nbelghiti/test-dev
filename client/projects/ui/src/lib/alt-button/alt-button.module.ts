import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';

import { IconModule } from '../icon/icon.module';

import { AltButtonComponent } from './alt-button.component';

@NgModule({
	imports: [
		CommonModule,
		IconModule,
		RouterModule,
	],
	declarations: [
		AltButtonComponent,
	],
	exports: [
		AltButtonComponent,
	],
})
export class AltButtonModule {}
