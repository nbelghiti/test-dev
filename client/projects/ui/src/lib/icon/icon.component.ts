import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
