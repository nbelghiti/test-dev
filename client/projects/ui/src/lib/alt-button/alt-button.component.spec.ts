import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { RouterTestingModule } from '@angular/router/testing';

import { IconComponent } from '../icon/icon.component';
import { IconName } from '../icon/icon.types';

import { AltButtonComponent } from './alt-button.component';

describe('AltButtonComponent', () => {
	let component: AltButtonComponent;
	let fixture: ComponentFixture<AltButtonComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [ RouterTestingModule ],
			declarations: [ AltButtonComponent, IconComponent ]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(AltButtonComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});

	describe('check inputs', () => {
		beforeEach(() => {
			fixture = TestBed.createComponent(AltButtonComponent);

			component = fixture.componentInstance;

			fixture.detectChanges();
		});

		it('should find iconBefore class when iconBefore input is passed', () => {
			component.iconBefore = IconName.book;

			fixture.detectChanges();

			expect(
				fixture.debugElement.query(By.css('.c-alt-button__icon--before'))
			).toBeTruthy();
			expect(
				fixture.debugElement.query(By.css('.c-alt-button__icon--after'))
			).toBeFalsy();
		});

		it('should find iconAfter class when iconAfter input is passed', () => {
			component.iconAfter = IconName.book;

			fixture.detectChanges();

			expect(
				fixture.debugElement.query(By.css('.c-alt-button__icon--after'))
			).toBeTruthy();
			expect(
				fixture.debugElement.query(By.css('.c-alt-button__icon--before'))
			).toBeFalsy();
		});

		it('should trigger clicked output when button is clicked', () => {
			let target = null;

			const buttonEl = fixture.nativeElement.firstElementChild;
			component.clicked.subscribe(val => target = val.target);

			buttonEl.click();

			expect(target).toEqual(buttonEl);
		});
	});

});
