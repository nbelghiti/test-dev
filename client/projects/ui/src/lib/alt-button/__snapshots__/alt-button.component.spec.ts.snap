// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AltButtonComponent Should render correctly 1`] = `
<ui-alt-button
  IconSize={[Function Object]}
  clicked={[Function EventEmitter]}
>
  <button
    class="c-alt-button"
  >
    
    
    <span
      class="c-alt-button__text"
    />
    
    
  </button>
</ui-alt-button>
`;
